openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -344,29 +345,75 @@ def post(self, request):
+                return verify_user_password_response
             with transaction.atomic():
                 # 1. Unlink LMS social auth accounts
-                UserSocialAuth.objects.filter(user_id=user.id).delete()
+                UserSocialAuth.objects.filter(user_id=request.user.id).delete()
                 # 2. Change LMS password & email
-                user.email = get_retired_email_by_email(user.email)
-                user.save()
-                _set_unusable_password(user)
+                request.user.email = get_retired_email_by_email(request.user.email)
+                request.user.save()
+                _set_unusable_password(request.user)
                 # 3. Unlink social accounts & change password on each IDA, still to be implemented
+                # 4. Add user to retirement queue
+                UserRetirementStatus.create_retirement(request.user)
+                # 5. Log the user out
+                logout(request)
             return Response(status=status.HTTP_204_NO_CONTENT)
         except KeyError:
             return Response(u'Username not specified.', status=status.HTTP_404_NOT_FOUND)
         except user_model.DoesNotExist:
-            return Response(u'The user "{}" does not exist.'.format(username), status=status.HTTP_404_NOT_FOUND)
+            return Response(
+                u'The user "{}" does not exist.'.format(request.user.username), status=status.HTTP_404_NOT_FOUND
+            )
         except Exception as exc:  # pylint: disable=broad-except
             return Response(text_type(exc), status=status.HTTP_500_INTERNAL_SERVER_ERROR)
 
+    def _verify_user_password(self, request):
+        """
+        If the user is logged in and we want to verify that they have submitted the correct password
+        for a major account change (for example, retiring this user's account).
+
+        Args:
+            request (HttpRequest): A request object where the password should be included in the POST fields.
+        """
+        try:
+            self._check_excessive_login_attempts(request.user)
+            user = authenticate(username=request.user.username, password=request.POST['password'], request=request)
+            if user:
+                if LoginFailures.is_feature_enabled():
+                    LoginFailures.clear_lockout_counter(user)
+                return Response(status=status.HTTP_204_NO_CONTENT)
+            else:
+                self._handle_failed_authentication(request.user)
+        except AuthFailedError as err:
+            return Response(text_type(err), status=status.HTTP_403_FORBIDDEN)
+        except Exception as err:  # pylint: disable=broad-except
+            return Response(u"Could not verify user password", status=status.HTTP_400_BAD_REQUEST)
+
+    def _check_excessive_login_attempts(self, user):
+        """
+        See if account has been locked out due to excessive login failures
+        """
+        if user and LoginFailures.is_feature_enabled():
```

Comment:
any reason not to combine these two boolean statements into one a la

```python
if user and LoginFailures.is_feature_enabled() and LoginFailures.is_user_locked_out(user):
  raise AuthFailedError()
```

I get that it's longer, but you can also put the `and`ed statements on new lines maybe?

Reply:
Because this is a purely stylistic decision I just went with what I think reads better; I don't like multi-line conditionals if I can avoid them.